### PR TITLE
New census summarization logic

### PR DESCRIPTION
**Pull request by drewsamnick**
This changes the census summarization logic to the simple logic in [the current spec](https://docs.google.com/document/d/1bdLSxePi0R4w1Q56VvQmv-TDgjVGa3Wsw09Nccmprp4/edit#). I've kept around the Bayesian summarization logic since we may still want to use it or a variation. 

There are a few test cases that are ambiguous depending on which logic we use. For now I've changed them to expect `MAYBE` or the previously expected value (`YES` or `NO`) so that they don't need to be tweaked every time we update the logic. Either result is appropriate in these case, but not the opposite value or nil so the tests are still useful.

`bin/oneoff/census/summarize.rb` has ben updated to output the summaries rather than save to the DB. It also now takes an optional parameter to control which algorithm to use. This will allow us to try different logic more easily against prod data.

I also added summarization to the cron job for updating the summaries fusion table so that summaries will be updated daily.

### CODE REVIEW

File: dashboard/app/models/census/census_summary.rb [hunk 1]
```diff
@@ -146,13 +147,138 @@ def self.decide_teaches(posterior)
   end
 
   def self.summarize_school_data(school, school_years, years_with_ap_data, years_with_ib_data, state_years_with_data)
+    summarize_school_data_simple(school, school_years, years_with_ap_data, years_with_ib_data, state_years_with_data)
```
Comment (breville) [inline dashboard/app/models/census/census_summary.rb]: Maybe worth passing a hash with this many parameters?

File: dashboard/app/models/census/census_summary.rb [hunk 2]
```diff
@@ -146,13 +147,138 @@ def self.decide_teaches(posterior)
...
+  end
+
+  def self.summarize_school_data_simple(school, school_years, years_with_ap_data, years_with_ib_data, state_years_with_data)
+    summaries = []
+
+    school_years.each do |school_year|
+      audit = {
+        version: 0.3,
```
Comment (breville) [inline dashboard/app/models/census/census_summary.rb]: Is 0.3 a meaningful constant that should be defined somewhere?

File: dashboard/app/models/census/census_summary.rb [hunk 3]
```diff
@@ -146,13 +147,138 @@ def self.decide_teaches(posterior)
...
+        ap_cs_offerings: [],
+        ib_cs_offerings: [],
+        state_cs_offerings: [],
+      }
+
+      count_yes = 0
+      count_no = 0
+
+      # If the school doesn't have stats then treat it as not high school.
+      # The lack of stats will show up in the audit data as a null value for high_school.
```
Comment (breville) [inline dashboard/app/models/census/census_summary.rb:188]: Same for K8 school, given the lines below that do the same thing as high school?

File: dashboard/app/models/school_stats_by_year.rb [hunk 4]
```diff
@@ -93,4 +93,8 @@ def self.merge_from_csv(filename, options = {col_sep: "\t", headers: true, quote
   def has_high_school_grades?
     grade_09_offered || grade_10_offered || grade_11_offered || grade_12_offered || grade_13_offered
   end
+
+  def has_k8_grades?
+    grade_kg_offered || grade_01_offered || grade_02_offered || grade_03_offered || grade_04_offered || grade_05_offered || grade_06_offered || grade_07_offered || grade_08_offered
```
Comment (breville) [inline dashboard/app/models/school_stats_by_year.rb:98]: I feel like Ruby must have some magic to generate this list programatically :)

Comment (aoby) [reply dashboard/app/models/school_stats_by_year.rb:98]: You could do something like this:
```ruby
irb(main):014:0> (['kg'] + (9..13).to_a).map{|n| "grade_#{n.to_s.rjust(2,'0')}_offered"}
=> ["grade_kg_offered", "grade_09_offered", "grade_10_offered", "grade_11_offered", "grade_12_offered", "grade_13_offered"]
```

Then you'll need to run those strings through `read_attribute` and `reduce` or something similar to `||` them together. I'm not sure that will be more readable.

File: dashboard/test/models/census/census_summary_test.rb [hunk 5]
```diff
@@ -39,72 +39,77 @@ class Census::CensusSummaryTest < ActiveSupport::TestCase
...
   test "Non-high school with 20 hours SOME does teach" do
     submission = build :census_submission, how_many_10_hours: "NONE", how_many_20_hours: "SOME"
-    assert Census::CensusSummary.submission_teaches_cs?(submission, is_high_school: false)
+    assert Census::CensusSummary.submission_teaches_cs?(submission, is_high_school: false, is_k8_school: true)
   end
 
   test "Non-high school with 20 hours ALL does teach" do
     submission = build :census_submission, how_many_10_hours: "NONE", how_many_20_hours: "ALL"
-    assert Census::CensusSummary.submission_teaches_cs?(submission, is_high_school: false)
+    assert Census::CensusSummary.submission_teaches_cs?(submission, is_high_school: false, is_k8_school: true)
```
Comment (aoby) [inline dashboard/test/models/census/census_summary_test.rb:112]: Is there a test for `is_k8_school: nil`? According to the comments above that is a valid scenario.

File: dashboard/app/models/census/census_summary.rb (hunk 1 above)
Comment (drewsamnick) [reply dashboard/app/models/census/census_summary.rb]: Changed

File: dashboard/app/models/census/census_summary.rb (hunk 2 above)
Comment (drewsamnick) [reply dashboard/app/models/census/census_summary.rb]: Added constants to define the different versions.

File: dashboard/app/models/census/census_summary.rb (hunk 3 above)
Comment (drewsamnick) [reply dashboard/app/models/census/census_summary.rb:188]: Updated the comment to mention k8_schools too.

File: dashboard/app/models/school_stats_by_year.rb (hunk 4 above)
Comment (drewsamnick) [reply dashboard/app/models/school_stats_by_year.rb:98]: Not that I could determine.

File: dashboard/test/models/census/census_summary_test.rb (hunk 5 above)
Comment (drewsamnick) [reply dashboard/test/models/census/census_summary_test.rb:112]: Good point. I'll add this.

Comment (drewsamnick) [reply dashboard/test/models/census/census_summary_test.rb:112]: Added.

**Pull request merged**